_posts/2024-05-28-edb-and-kasten.markdown

```diff
@@ -0,0 +1,295 @@
+
+![Before snapshot](/images/posts/2024-05-28-edb-and-kasten/before-snapshot.png)
+
+When an intance is "Fenced" then it is garanteed to be consistent, which means that all transaction has been properly commited before the fence command
+return so now Kasten is 100% sure to take a fully consistent snapshot of the database.
+
+When the snapshot of the "Fenced" instance is over then Kasten will "Unfence" it and the instance will catch up with the last transactions that he eventually missed during the fencing.
+
+![After snapshot](/images/posts/2024-05-28-edb-and-kasten/after-snapshot.png)
+
+Now when it comes to restore, Kasten will only restore the PVC that was fenced and the operator will promote this instance primary 
+and recreate from it two replicas.
+
+![At restore](/images/posts/2024-05-28-edb-and-kasten/at-restore.png)
+
+# Let's try it now 
+
+## Install the operator 
+
+If you already have EDB operator installed on kubernetes you can skip this part  
+
+```
+kubectl apply -f https://get.enterprisedb.io/cnp/postgresql-operator-1.20.2.yaml
+```
+
+This will create the operator namespace where the controller will be running.
+
+## Create an EDB cluster, a client and some data 
+
+
+```
+kubectl create ns edb
+kubectl apply -f https://github.com/michaelcourcy/edb-kasten/raw/main/cluster-example-2.yaml -n edb
+```
+
+Wait for the cluster to be fully ready.
+```
+kubectl get clusters.postgresql.k8s.enterprisedb.io -n edb
+NAME              AGE   INSTANCES   READY   STATUS                     PRIMARY
+cluster-example   19m   3           3       Cluster in healthy state   cluster-example-1
+```
+
+
+Install the cnp plugin if you haven't it yet 
+```
+curl -sSfL \
+  https://github.com/EnterpriseDB/kubectl-cnp/raw/main/install.sh | \
+  sudo sh -s -- -b /usr/local/bin
+```
+
+Create a client certificate to the database
+```
+kubectl cnp certificate cluster-app \
+  --cnp-cluster cluster-example \
+  --cnp-user app \
+  -n edb 
+```
+
+Now you can create the client 
+```
+kubectl create -f https://github.com/michaelcourcy/edb-kasten/raw/main/client.yaml -n edb 
+```
+
+Create some data 
+```
+kubectl exec -it deploy/cert-test -- bash
+psql " $DATABASE_URL "
+\c app
+DROP TABLE IF EXISTS links;
+CREATE TABLE links (
+	id SERIAL PRIMARY KEY,
+	url VARCHAR(255) NOT NULL,
+	name VARCHAR(255) NOT NULL,
+	description VARCHAR (255),
+        last_update DATE
+);
+INSERT INTO links (url, name, description, last_update) VALUES('https://kasten.io','Kasten','Backup on kubernetes',NOW());
+select * from links;
+\q
+exit
+```
+
+## Add the backup decorator annotations to the cluster 
+
+You can skip this part if you create the cluter from the previous section because with [cluster-example-2](https://github.com/michaelcourcy/edb-kasten/raw/main/cluster-example-2.yaml) the cluster-example already include the kasten addon.
```

Comment:
Skip this step if you have created the cluster using the instructions from the previous section. With [cluster-example-2](https://github.com/michaelcourcy/edb-kasten/raw/main/cluster-example-2.yaml) the yaml includes the kasten addon.